### PR TITLE
chore(storage): Add tests to cover empty string/file upload

### PR DESCRIPTION
**Pull request by bajajneha27**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-storage/test/google/cloud/storage/bucket_test.rb [hunk 1]
```diff
@@ -581,6 +581,41 @@
     end
   end
 
+  it "creates an empty file" do
+    new_file_name = random_file_path
+
+    Tempfile.open ["google-cloud", ".txt"] do |tmpfile|
+      tmpfile.write ""
+      tmpfile.rewind
```
Comment (dazuma) [inline google-cloud-storage/test/google/cloud/storage/bucket_test.rb:589]: I don't think these two lines are necessary. Either `Tempfile.open` or `Tempfile.create` will create an empty file.

File: google-cloud-storage/test/google/cloud/storage/bucket_test.rb [hunk 2]
```diff
@@ -581,6 +581,41 @@
     end
   end
 
+  it "creates an empty file" do
+    new_file_name = random_file_path
+
+    Tempfile.open ["google-cloud", ".txt"] do |tmpfile|
```
Comment (dazuma) [inline google-cloud-storage/test/google/cloud/storage/bucket_test.rb:587]: Prefer `Tempfile.create` rather than `Tempfile.open`.

File: google-cloud-storage/test/google/cloud/storage/bucket_test.rb [hunk 3]
```diff
@@ -581,6 +581,41 @@
...
+      created = bucket_user_project.create_file tmpfile, new_file_name
+      _(created.user_project).must_equal true
+
+      mock.verify
+    end
+  end
+
+  it "creates an file with a StringIO" do
+    new_file_name = random_file_path
+    new_file_contents = StringIO.new ""
```
Comment (dazuma) [inline google-cloud-storage/test/google/cloud/storage/bucket_test.rb:606]: The argument is unnecessary.

File: google-cloud-storage/test/google/cloud/storage/lazy/bucket_test.rb [hunk 4]
```diff
@@ -336,6 +336,27 @@
     end
   end
 
+  focus
```
Comment (dazuma) [inline google-cloud-storage/test/google/cloud/storage/lazy/bucket_test.rb:339]: Remove the focus

File: google-cloud-storage/test/google/cloud/storage/lazy/bucket_test.rb [hunk 5]
```diff
@@ -336,6 +336,27 @@
...
+  it "creates an empty file" do
+    new_file_name = random_file_path
+
+    Tempfile.open ["google-cloud", ".txt"] do |tmpfile|
+      tmpfile.write ""
+      tmpfile.rewind
```
Comment (dazuma) [inline google-cloud-storage/test/google/cloud/storage/lazy/bucket_test.rb:345]: See earlier comments about Tempfile.

**Review by dazuma — CHANGES_REQUESTED** [review]

File: google-cloud-storage/acceptance/storage/file_test.rb [hunk 6]
```diff
@@ -457,6 +457,23 @@
...
+    _(uploaded.name).must_equal "uploaded/empty-file.txt"
+
+    downloadio = StringIO.new
+    downloaded = uploaded.download downloadio
+    _(downloaded).must_be_kind_of StringIO
+
+    downloaded_data = downloaded.string
+    _(downloaded_data).must_equal data
+    ensure
+      uploaded.delete
```
Comment (dazuma) [inline google-cloud-storage/acceptance/storage/file_test.rb:474]: The above two lines should probably be outdated 2 spaces.

**Review by dazuma — APPROVED** [review]
LGTM
